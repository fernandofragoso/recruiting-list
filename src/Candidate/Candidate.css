.Candidate {
  display: flex;
  border: 1px solid #CCC;
  margin: 10px;
  padding: 10px;
}

.Candidate__left {
  margin: 10px 0 0 10px;
  flex: 0 0 150px;
  text-align: left;
}

.Candidate__center {
  flex: 1 0 auto;
  text-align: left;
  overflow: hidden;
}

.Candidate__right {
  margin: 10px 10px 0 0;
  flex: 0 0 100px;
  display: flex;
  text-align: right;
}

.Candidate__image {
  object-fit: cover;
  width:120px;
  height:120px;
}