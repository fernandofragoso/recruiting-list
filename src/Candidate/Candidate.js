import React, { Component } from 'react';
import './Candidate.css';

import CandidateInfo from './CandidateInfo/CandidateInfo';

class Candidate extends Component {
  
  constructor(props) {
    super(props);
  }
  
  render() {
    return (
      <div className="Candidate">
        <section className="Candidate__left">
          <img className="Candidate__image" alt="Candidate" src={this.props.candidate.imageUrl} />
        </section>
        <section className="Candidate__center">
          <CandidateInfo candidate={this.props.candidate} />
        </section>
        <section className="Candidate__right">
        </section>
        
      </div>
    );
  }
}

export default Candidate;
