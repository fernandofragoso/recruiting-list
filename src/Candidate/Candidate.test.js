import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Candidate from './Candidate';

let candidate = {
  "id": 3,
  "imageUrl":
    "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFWfkA6fPozy8qlbL8q282B2jWo10s-UK2bYFLv1b_v32r5TAF",
  "name": "Raoul Marquardt",
  "newThisWeek": true,
  "career": "Chief Security Representative",
  "lastCompanies": [
    "Thompson, Hintz and Walker",
    "Aufderhar, Goldner and Dach",
    "Predovic, Gutkowski and Lubowitz"
  ],
  "university": "Kentucky",
  "cities": ["Lake Bridgette", "Abeton"],
  "mainSkills": ["Creative", "Solutions"]
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Candidate candidate={candidate} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders and match snapshot', () => {
  const tree = renderer.create(<Candidate candidate={candidate} />);
  expect(tree).toMatchSnapshot();
});