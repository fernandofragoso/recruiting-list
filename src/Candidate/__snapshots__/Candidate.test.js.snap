// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and match snapshot 1`] = `
<div
  className="Candidate"
>
  <section
    className="Candidate__left"
  >
    <img
      alt="Candidate"
      className="Candidate__image"
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTFWfkA6fPozy8qlbL8q282B2jWo10s-UK2bYFLv1b_v32r5TAF"
    />
  </section>
  <section
    className="Candidate__center"
  >
    <div
      className="CandidateInfo"
    >
      <h3>
        Raoul Marquardt
         
        <span
          className="CandidateInfo__pill CandidateInfo__pill--new"
        >
          Novo essa semana
        </span>
      </h3>
      <h4>
        Chief Security Representative
      </h4>
      <div
        className="CandidateInfo__item"
      >
        <img
          alt="Work icon"
          className="icon"
          src="work-icn.svg"
        />
        Thompson, Hintz and Walker, Aufderhar, Goldner and Dach, Predovic, Gutkowski and Lubowitz
      </div>
      <div
        className="CandidateInfo__item"
      >
        <img
          alt="Study icon"
          className="icon"
          src="study-icn.svg"
        />
        Kentucky
      </div>
      <div
        className="CandidateInfo__item"
      >
        <img
          alt="Location icon"
          className="icon"
          src="location-icn.svg"
        />
        <span>
          Disposto a trabalhar em: 
        </span>
        Lake Bridgette, Abeton
      </div>
      <div
        className="CandidateInfo__item"
      >
        <img
          alt="Skill icon"
          className="icon"
          src="skills-icn.svg"
        />
        <span>
          Principais habilidades: 
        </span>
        <span
          className="CandidateInfo__pill CandidateInfo__pill--skill"
        >
          Creative
        </span>
        <span
          className="CandidateInfo__pill CandidateInfo__pill--skill"
        >
          Solutions
        </span>
      </div>
    </div>
  </section>
  <section
    className="Candidate__right"
  >
    <div
      className="CandidateActions"
    >
      <button
        className="CandidateActions__button"
        onClick={[Function]}
      >
        <img
          src="star2.svg"
        />
      </button>
      <div
        className="CandidateActions__popover-wrapper"
      >
        <button
          className="CandidateActions__button"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <img
            src="dots.svg"
          />
        </button>
        <div
          className="CandidateActions__popover-button"
        >
          <button
            className="CandidateActions__button CandidateActions__button--popover"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            Remover
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
